Add extraction of high-quality docs to text files

MarekGarbowski opened the pull request:
This commit introduces a new function save_high_quality_docs() in the extraction_to_txt_file.py. It saves high-quality documents from the Speakleash dataset to a specified folder, creating the necessary directories if they don't exist. The .gitignore file has been updated to exclude the output directory from the repository. This function helps in segregating high quality documents for further analysis.

Commit:
Add extraction of high-quality docs to text files

This commit introduces a new function save_high_quality_docs() in the extraction_to_txt_file.py. It saves high-quality documents from the Speakleash dataset to a specified folder, creating the necessary directories if they don't exist. The .gitignore file has been updated to exclude the output directory from the repository. This function helps in segregating high quality documents for further analysis.

IgorTest19 commented on example_3_quality_metrics/extraction_to_txt_file.py:
```diff
@@ -0,0 +1,44 @@
+import os
```
Missing module docstring.

IgorTest19 commented on example_3_quality_metrics/extraction_to_txt_file.py:
```diff
@@ -0,0 +1,44 @@
+from speakleash import Speakleash
+
+
+def save_high_quality_docs(txt_folder_name):
```
You can split up functionalities using 2 functions:
- `def get_datatxt_folder_name)` which creates directory
- `def save_quality_docs(quality='HIGH')` which will wrap code from creating an instance of the Speakleash class (from line 26 to 41)
- use these two functions inside of condition:` if __name__ == '__main__':`

Look at the following example:
https://github.com/speakleash/speakleash-examples/pull/13/files (file charts.py)

IgorTest19 commented on .gitignore:
```diff
@@ -157,6 +157,7 @@ cython_debug/
 #  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
```
Remove .gitignore file. It's not needed there :)

Commit:
Update extraction_to_txt_file script for versatility

The extraction_to_txt_file script has been revised for added versatility. A new module has been added providing functions to create necessary directories and save documents from the 'speakleash' dataset to a specific folder. The 'save_high_quality_docs' method was updated to 'save_quality_docs' to allow for selection of document quality. New usage information was also added for guidance on the script modification, importing and using the module's functions, running the script directly, and the output from the script execution. A 'get_data' function was also added to manage the creation of necessary directories. This update provides improved script functionality and versatile document retrieval.

MarekGarbowski commented:
The extraction_to_txt_file script has been revised for added versatility. A new module has been added providing functions to create necessary directories and save documents from the 'speakleash' dataset to a specific folder. The 'save_high_quality_docs' method was updated to 'save_quality_docs' to allow for selection of document quality. New usage information was also added for guidance on the script modification, importing and using the module's functions, running the script directly, and the output from the script execution. A 'get_data' function was also added to manage the creation of necessary directories. This update provides improved script functionality and versatile document retrieval.

Commit:
Update docstring position in extraction_to_txt_file methods

The commit relocates the docstrings to be directly under the method definition, as per Python's standard practice, in the 'extraction_to_txt_file' script. These provide information about the functions 'get_data' and 'save_quality_docs'. This revision ensures that the method's purpose is clarified to the developers in the correct position as a part of Python's best practices. This doesn't affect functionality but improves script readability.
Removed .gitignore file from repo

Commit:
Update docstring position in extraction_to_txt_file methods

The commit relocates the docstrings to be directly under the method definition, as per Python's standard practice, in the 'extraction_to_txt_file' script. These provide information about the functions 'get_data' and 'save_quality_docs'. This revision ensures that the method's purpose is clarified to the developers in the correct position as a part of Python's best practices. This doesn't affect functionality but improves script readability.
Removed .gitignore file from repo

Commit:
Commit message:
    Relocate imports in extraction_to_txt_file.py

This commit moves import statements (os and speakleash) to after module-level docstring in extraction_to_txt_file.py.

Commit:
Add README.md for "extraction_to_files" module, refactor related script, and move files to a separate folder.

MarekGarbowski commented:
Added README.md file and both moved to separate folder.

Commit:
Modified the readme.md and main files as suggested.

MarekGarbowski commented:
Modified the readme.md and main files as suggested.

IgorTest19 committed:
Update README.md
